C#/Botkit/Microsoft.BotKit.Adapters.Slack/SlackAdapter.cs

```diff
@@ -248,15 +264,15 @@ public override async Task<ResourceResponse[]> SendActivitiesAsync(ITurnContext
         /// <param name="activity">The updated activity in the form `{id: <id of activity to update>, ...}`</param>
         public override async Task<ResourceResponse> UpdateActivityAsync(ITurnContext turnContext, Activity activity, CancellationToken cancellationToken)
         {
-            ResourceResponse results = null;
+            dynamic results = null;
             if (activity.Id != null && activity.Conversation != null)
             {
                 try
                 {
                     dynamic message = ActivityToSlack(activity);
-                    SlackAPI slack = await GetAPIAsync(activity);
-                    //results = await slack.chat.update(message);
-                    if (/*!results.ok*/ true)
+                    SlackTaskClient slack = await GetAPIAsync(activity);
+                    results = await slack.UpdateAsync(activity.Timestamp.ToString(), activity.ChannelId, message);
+                    if (!results.ok)
                     {
                         Console.WriteLine("Error updating activity on Slack:", results);
```

Comment:
In this case, after removing the previous `try/catch`, replace this `Console.WriteLine` with a `throw new Exception` with the same error message.